lib/modules/dosomething/dosomething_global/dosomething_global.module

```diff
@@ -58,9 +58,14 @@ function dosomething_global_init() {
   // Handle translation redirects for authenticated users when they request
   // bare (Global English) URLs to user pages (user profile, edit, etc.).
   //
+  // Don't redirect POSTs.
+  //
   // Passing 'noredirect=1' will defeat the redirect.
-  if (arg(0) == 'user' && module_exists('dosomething_global') &&
-      empty($_GET['noredirect'])) {
+
+  $request_method = strtolower($_SERVER['REQUEST_METHOD']);
+
+  if (arg(0) == 'user' && module_exists('dosomething_global') && 'post' !== $request_method &&
```

Comment:
Are you sure we shouldn't issue redirects only for `/user/ *`? Mayne let's skip it for all POSTs so post data won't get lost?